subprojects/packagefiles/luajit/meson.build

```diff
@@ -1,18 +1,47 @@
 project('luajit', 'c', version : '2.0.5', license : 'mit')
 
 # TODO: This is not ideal. We should translate the enough of the makefile to have Meson handle the build.
+cc = meson.get_compiler('c')
+
+env = ['DESTDIR="' + (meson.current_source_dir() / 'out') + '"']
+if meson.is_cross_build()
+    # Construct the compiler prefix, e.g. 'x86_64-w64-mingw32-'.
+    # FIXME: This is absolutely disgusting. I'm not sure whether to loathe myself or blame the Meson team.
+    cc_name_parts = cc.cmd_array()[0].split('/')[-1].split('\\')[-1].split('-')
+    cc_name_hyphen = ''
+    foreach cc_name_part: cc_name_parts
+        cc_name_prefix = cc_name_hyphen
+        cc_name_hyphen += cc_name_part + '-'
+    endforeach
+
+    env += ['CROSS=' + cc_name_prefix]
+    if host_machine.system() == 'darwin'
+        env += 'TARGET_SYS=Darwin'
+    elif host_machine.system() == 'linux'
+        env += 'TARGET_SYS=Linux'
+    elif host_machine.system() == 'windows'
+        env += 'TARGET_SYS=Windows'
+    endif
+endif
+
 run_command(
-    'make', '-C', meson.current_source_dir(), 'install',
+    'make', '-C', meson.current_source_dir(), 'amalg',
     check: true,
-    env: ['DESTDIR="' + (meson.current_source_dir() / 'out') + '"'])
+    env: env)
 
-cc = meson.get_compiler('c')
-libs = cc.find_library(
-    'luajit-5.1',
-    dirs: meson.current_source_dir() / 'out/usr/local/lib',
-    static: true)
+if host_machine.system() == 'windows'
```

Comment:
What about other windows environments like cygwin?

Reply:
Yeah, I'm concerned about that. (Not a Microsoft customer, can't test locally.) My educated guess is that the Meson subproject never worked there in the first place. So far, the Windows CI builds install LuaJIT via MSYS2 and bypass all of this. They end up using the dynamic library, lua51.dll.
The solution might be another CI environment. Would need to discuss if that's worth it.